File: anul2/semestrul1/map/lab2/model/AbstractContainer.java
package model;

import java.util.ArrayList;
import java.util.List;

public abstract class AbstractContainer implements Container {
    protected List<Task> tasks = new ArrayList<>();

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }

    @Override
    public void add(Task task){
        tasks.add(task);
    }

    @Override
    //dar e clasa abstracta ce trebuie implementata de containere
    public abstract Task remove();
}

File: Third Semester - Advanced Programming Methods/Lab10/model/AbstractContainer.cs
namespace lab10.model;

public abstract class AbstractContainer:IContainer
{
    protected List<Task> tasks = new List<Task>();

    public abstract Task Remove();

    public void Add(Task task)
    {
        tasks.Add(task);
    }

    public int Size()
    {
        return tasks.Count;
    }

    public bool IsEmpty()
    {
        return tasks.Count == 0;
    }

    public List<Task> Container
    {
        get;
        set;
    }
}